src/metabase/api/user.clj

```diff
@@ -299,16 +299,13 @@
   "True when the user has permissions for at least one un-archived question and one un-archived dashboard, excluding
   internal/automatically-loaded content."
   [user]
-  (let [coll-ids-filter (collection/visible-collection-ids->honeysql-filter-clause
-                         :collection_id
-                         (collection/permissions-set->visible-collection-ids @api/*current-user-permissions-set*))
-        entity-exists? (fn [model] (t2/exists? model
-                                               {:where [:and
-                                                        [:= :archived false]
-                                                        coll-ids-filter
-                                                        (mi/exclude-internal-content-hsql model)]}))]
```

Comment:
Looks like you might have lost the `:= :archived false` and `exclude-internal-content-hsql` clauses here?